subsys/rtio/rtio_userspace_init.c

```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/app_memory/app_memdomain.h>
+
+K_APPMEM_PARTITION_DEFINE(rtio_partition);
```

Comment:
I think this could be made read only to user space threads and increase the safety around this memory. The management data (mempool_map) and the data itself (array) are intended for read only usage from the user space thread as far as I can see.